shell/platform/darwin/macos/framework/Source/FlutterMetalTextureProvider.mm

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterMetalTextureProvider.h"
+
+#include <Metal/Metal.h>
+
+@interface FlutterMetalTextureProvider () {
+  id<MTLDevice> _device;
+}
+@end
+
+@implementation FlutterMetalTextureProvider
+- (instancetype)initWithMTLDevice:(const id<MTLDevice>)device {
+  if (self = [super init]) {
+    _device = device;
+  }
+  return self;
+}
+
+- (id<MTLTexture>)createTextureWithSize:(CGSize)size iosurface:(nonnull IOSurfaceRef) ioSurface {
+  auto texture_descriptor = [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm
```

Comment:
we have similar functionality in `FlutterSurfaceManager` as well, can we unify?

Reply:
I was thinking some more about this. I'm not sure it's worth unifying FlutterSurfaceManager with this as it's literally only two lines of code, but I think I will get rid of FlutterMetalTextureProvider and just make the texture in FlutterMetalCompositor. Two lines of code doesn't seem worth the complexity of refactoring.